Makefile

```diff
@@ -50,6 +50,9 @@ SPKG_COLLECT_FILES = build/pkgs/*/type build/pkgs/*/package-version.txt build/pk
 # If configure was run before, rerun it with the old arguments.
 # Otherwise, run configure with argument $PREREQ_OPTIONS.
 build/make/Makefile: configure $(SPKG_COLLECT_FILES) $(CONFIG_FILES:%=%.in)
+	$(MAKE) reconfigure
+
+reconfigure:
 	rm -f config.log
 	mkdir -p logs/pkgs
 	ln -s logs/pkgs/config.log config.log
```

Comment:
You didn't touch lines 56--58, but now `make reconfigure` runs these lines repeatedly. I wonder if this is really necessary. What is the purpose of these lines?

Reply:
I experimented with it. It seems that removing the three lines does not do any harm, and we can keep the log in the root directory...

Reply:
I don't understand what you mean. There was no `make reconfigure` target before this PR. 
The rule `build/make/Makefile` used to run lines 56–58 directly.
Now it runs them by delegating to `make reconfigure`

Reply:
The lines are not run by the first run of `./configure`. Hence we have the file `config.log` at the root directory. The old message from `./configure` says 
```
    hint: After installation, re-run configure using:

      $ ./config.status --recheck && ./config.status
```
Hence re-running the above code does not create the symbolic link to the log file in `logs/pkgs/config.log`.

I meant that with your PR, the lines are run by `make reconfigure`, and we have the symbolic link.

Reply:
Oh, now I understand.

Reply:
At least I think sio. You mean that if the user just runs `./config.status --recheck && ./config.status` (as previously instructed by `configure`), the linking is not done, and now with the replacement `make reconfigure` it is done?

Reply:
Yes. And if I run `make reconfigure` again, it recreates the symbolic link.

Reply:
Above all, I see no reason to keep `config.log` in `logs/pkgs`.

Reply:
Neither do I, but it has been like that since at least 2015 (0e168b13f36e0b6af6de5d521fa18f9054cb1add), and I don't want to change it here.

Reply:
In #33262, I had an improvement for the handling of config.log though, please take a look

Reply:
Ah, I now see that the symlink is created anyway by the target `build/make/Makefile`, and we end up with the symlink at the end of installing sage, though not immediately after `./configure`. Hence there is no change in this regard with this PR. OK then.

I still think the directory `logs/pkgs` is not the right place for `config.log` though. Anyway this is not a concern of this PR.

OK. Thanks.

Reply:
> In #33262, I had an improvement for the handling of config.log though, please take a look

That's now revived as #36998